sklearn/utils/deprecation.py

```diff
@@ -76,6 +79,22 @@ def wrapped(*args, **kwargs):
 
         return wrapped
 
+    def _decorate_instance(self, instance):
+        cls = instance.__class__
+        msg = "Thing %s is deprecated" % cls.__name__
```

Comment:
Because I am not at all sure that we want to use `_decorate_instance`, wanted to put it out there as an idea: https://github.com/scikit-learn/scikit-learn/pull/7261#discussion_r76738685

(conclusion: discussion in line comments isn't very findable)